### PR TITLE
const generics feature update internals post

### PR DESCRIPTION
**Pull request by lcnr**
explains the removal of `feature(const_generics)` and presents the currently existing feature gates.

should only be published after https://github.com/rust-lang/rust/pull/88369 has been merged

cc @rust-lang/project-const-generics

### CODE REVIEW

File: posts/inside-rust/2021-08-27-Splitting-const-generics.md
Comment (BoxyUwU) [inline posts/inside-rust/2021-08-27-Splitting-const-generics.md:53]:
```suggestion
Without any unstable features, const arguments must either be a fully concrete expression or a generic parameter by itself, so constants like `N + 1` are forbidden. With this feature, expressions using generic parameters are possible. 
```

File: posts/inside-rust/2021-08-27-Splitting-const-generics.md
Comment (BoxyUwU) [inline posts/inside-rust/2021-08-27-Splitting-const-generics.md:102]: hmm why is this feature called generic_arg_infer shouldnt it mention `const` somewhere :sweat_smile:

Comment (lcnr) [reply posts/inside-rust/2021-08-27-Splitting-const-generics.md:104]: because it adds a `hir::GenericArg::Infer` :laughing:

maybe `const_argument_wildcard` or something like that would be better :shrug: especially if we allow `[u8; _]` at the same time

File: posts/inside-rust/2021-08-27-Splitting-const-generics.md
Comment (lcnr) [reply posts/inside-rust/2021-08-27-Splitting-const-generics.md]: > I guess implementation-wise, routing Bar from Foo<Bar, {expr}> into the argument to type-check it as expr: Bar, is going to be a lot of the work necessary?

yeah, we could change https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.WithOptConstParam.html to store the `typecheck_type` instead of a param def id but that seems fragile to me

File: posts/inside-rust/2021-08-27-Splitting-const-generics.md
Comment (BoxyUwU) [reply posts/inside-rust/2021-08-27-Splitting-const-generics.md:104]: ah i guess it makes sense then :)

File: posts/inside-rust/2021-08-27-Splitting-const-generics.md
Comment (lqd) [reply posts/inside-rust/2021-08-27-Splitting-const-generics.md]: probably with the US spelling, "stabilization", as it is used in the rest of the document

File: posts/inside-rust/2021-08-27-Splitting-const-generics.md
Comment (lcnr) [inline posts/inside-rust/2021-08-27-Splitting-const-generics.md:44]:
```suggestion
There are still two major blockers for stabilization: 
```

**Comment by lcnr** [comment]
rust-lang/rust#88369 has been merged so the feature rename is going to be on nightly tomorrow.

It would be great to publish this once that nightly has been released. If this doesn't seem right for the internals blog I am going to instead post this to the const generics tracking issue.

**Comment by lcnr** [comment]
@rust-lang/core I would like to get this post published today, either here or as a comment on the const generics tracking issue, depending on what you prefer.

File: posts/inside-rust/2021-08-27-Splitting-const-generics.md
Comment (eddyb) [reply posts/inside-rust/2021-08-27-Splitting-const-generics.md]: `const_wildcard` maybe? Or `const_infer`. Ideally it would apply to all type-level constants.

**Review by steveklabnik — APPROVED** [review]
This looks good to me!

I will say one thing: it's very hard to post something and ask for a review within the same day. Most of the core team is actually asleep at this point, so it's unlikely that turnaround times are that fast. That also being said, I do appreciate being looped in, but inside rust is for the teams to own; don't feel you need our approval to post things here :)

File: posts/inside-rust/2021-09-01-Splitting-const-generics.md
Comment (calebcartwright) [inline posts/inside-rust/2021-09-01-Splitting-const-generics.md:51]: This may just be me reading something that isn't there so feel free to disregard, but the change from the plural to singular first person narrator here jumped out to me while reading. Wasn't sure whether this was reflective of there being differing opinions about this within the WG, or if it was just an accidental `I` instead of `we`

Comment (lcnr) [reply posts/inside-rust/2021-09-01-Splitting-const-generics.md]: true, this started out as a more personal blog and I didn't change all the "I" here. Changed it. While this is mostly reflective of my own opinion, I don't think the rest of @rust-lang/project-const-generics has vastly different expectations here

**Comment by lcnr** [comment]
> I will say one thing: it's very hard to post something and ask for a review within the same day. Most of the core team is actually asleep at this point, so it's unlikely that turnaround times are that fast.

:sweat_smile: I did not expect this to be reviewed in one day, as I posted this 5 days before that core ping. I assumed however that core already got a notification when this PR was opened so I did not want to ping you about it again without actually "needing to do anything". 

> That also being said, I do appreciate being looped in, but inside rust is for the teams to own; don't feel you need our approval to post things here :)

That confuses me a bit, as I assumed that core's approval is needed to merge anything here. I (as a project-lead) am not able to merge this PR by myself (if I wanted to), are compiler/libs-team leads able to do so? Who should I bother the next time I want to publish a blog post about this?

**Comment by pietroalbini** [comment]
@lcnr if you change the blog post date to today I can publish it later in the day!

**Comment by lcnr** [comment]
@pietroalbini done

**Review by pietroalbini — APPROVED** [review]

**Comment by pietroalbini** [comment]
Thanks, and sorry for the delay in merging this!

**Pull request merged**
